main.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+  "fmt"
+  "errors"
+  "unicode/utf8"
+)
+func main() {
+  // Initialise the input string
+  input := "The quick brown fox jumped over the lazy dog"
+  // Reverse input
+  reverse, reverseErr := Reverse(input)
+  doubleReverse, doubleReverseErr := Reverse(reverse)
```

Comment:
If a function call could return an err as well as some value, then you shouldn't be using the value before you've checked that the error is `nil`.
In this case you're working with strings, so it's almost certainly safe, but many other times you'd risk acting on a `nil` value, which could crash your app.